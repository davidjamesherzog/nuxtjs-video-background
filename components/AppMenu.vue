<template>
  <div class="menu" data-app-menu>
    <ul>
      <li>
        <a href="https://videojs.com/getting-started" data-app-menu-get-started
          >Get Started</a
        >
      </li>
      <li>
        <a
          href="https://videojs.com/getting-started/#customize"
          data-app-menu-customize
          >Customize</a
        >
      </li>
      <li><a href="https://docs.videojs.com/" data-app-menu-docs>Docs</a></li>
      <li><a href="https://videojs.com/blog" data-app-menu-blog>Blog</a></li>
      <li>
        <a href="https://github.com/videojs/video.js" data-app-menu-github
          >Github</a
        >
      </li>
    </ul>
  </div>
</template>

<style lang="scss" scoped>
.menu {
  position: absolute;
  top: 0;
  right: 0;
  width: 300px;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;

  ul {
    position: relative;

    li {
      list-style: none;

      a {
        text-decoration: none;
        font-size: 24px;
        color: $color-brand-black;

        &:hover {
          color: $color-brand-blue;
        }
      }
    }
  }
}
</style>
