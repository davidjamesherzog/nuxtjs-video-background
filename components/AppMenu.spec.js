import { mount } from '@vue/test-utils';
import AppMenu from '@/components/AppMenu.vue';

describe('AppMenu', () => {
  describe('templates', () => {
    test('is a Vue instance', () => {
      const wrapper = mount(AppMenu);
      expect(wrapper.vm).toBeTruthy();
      expect(wrapper.element).toMatchSnapshot();
    });

    test('should create menu', () => {
      const wrapper = mount(AppMenu);
      const menu = wrapper.find('[data-app-menu]');
      const getStarted = wrapper.find('[data-app-menu-get-started]');
      const customize = wrapper.find('[data-app-menu-customize]');
      const docs = wrapper.find('[data-app-menu-docs]');
      const blog = wrapper.find('[data-app-menu-blog]');
      const github = wrapper.find('[data-app-menu-github]');

      expect(menu.exists()).toBeTruthy();
      expect(menu.classes()).toContain('menu');
      expect(getStarted.exists()).toBeTruthy();
      expect(getStarted.text()).toContain('Get Started');
      expect(getStarted.attributes().href).toEqual(
        'https://videojs.com/getting-started'
      );
      expect(customize.exists()).toBeTruthy();
      expect(customize.text()).toContain('Customize');
      expect(customize.attributes().href).toEqual(
        'https://videojs.com/getting-started/#customize'
      );
      expect(docs.exists()).toBeTruthy();
      expect(docs.text()).toContain('Docs');
      expect(docs.attributes().href).toEqual('https://docs.videojs.com/');
      expect(blog.exists()).toBeTruthy();
      expect(blog.text()).toContain('Blog');
      expect(blog.attributes().href).toEqual('https://videojs.com/blog');
      expect(github.exists()).toBeTruthy();
      expect(github.text()).toContain('Github');
      expect(github.attributes().href).toEqual(
        'https://github.com/videojs/video.js'
      );
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
