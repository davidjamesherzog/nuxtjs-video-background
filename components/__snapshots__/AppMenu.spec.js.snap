// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppMenu templates is a Vue instance 1`] = `
<div
  class="menu"
  data-app-menu=""
>
  <ul>
    <li>
      <a
        data-app-menu-get-started=""
        href="https://videojs.com/getting-started"
      >
        Get Started
      </a>
    </li>
     
    <li>
      <a
        data-app-menu-customize=""
        href="https://videojs.com/getting-started/#customize"
      >
        Customize
      </a>
    </li>
     
    <li>
      <a
        data-app-menu-docs=""
        href="https://docs.videojs.com/"
      >
        Docs
      </a>
    </li>
     
    <li>
      <a
        data-app-menu-blog=""
        href="https://videojs.com/blog"
      >
        Blog
      </a>
    </li>
     
    <li>
      <a
        data-app-menu-github=""
        href="https://github.com/videojs/video.js"
      >
        Github
      </a>
    </li>
  </ul>
</div>
`;

exports[`AppMenu templates should create menu 1`] = `
<div
  class="menu"
  data-app-menu=""
>
  <ul>
    <li>
      <a
        data-app-menu-get-started=""
        href="https://videojs.com/getting-started"
      >
        Get Started
      </a>
    </li>
     
    <li>
      <a
        data-app-menu-customize=""
        href="https://videojs.com/getting-started/#customize"
      >
        Customize
      </a>
    </li>
     
    <li>
      <a
        data-app-menu-docs=""
        href="https://docs.videojs.com/"
      >
        Docs
      </a>
    </li>
     
    <li>
      <a
        data-app-menu-blog=""
        href="https://videojs.com/blog"
      >
        Blog
      </a>
    </li>
     
    <li>
      <a
        data-app-menu-github=""
        href="https://github.com/videojs/video.js"
      >
        Github
      </a>
    </li>
  </ul>
</div>
`;
